lisp/mastodon-auth.el

```diff
@@ -53,23 +53,21 @@
                        :host mastodon-instance-url
                        :port 443
                        :require '(:user :secret)))))
-    (let ((buffer
-           (mastodon-http--post
-            (concat mastodon-instance-url "/oauth/token")
-            `(("client_id" . ,(plist-get (mastodon-client) :client_id))
-              ("client_secret" . ,(plist-get (mastodon-client) :client_secret))
-              ("grant_type" . "password")
-              ("username" . ,(plist-get auth :user))
-              ("password" . ,(let ((secret (plist-get auth :secret)))
-                               (if (functionp secret)
-                                   (funcall secret)
-                                 secret)))
-              ("scope" . "read write follow"))
-            nil)))
+    (prog1
```

Comment:
Is there any chance that `auth-source-search` could return `nil` and therefore `auth` be nil as well?
If so, consider handling that more gracefully here since surely the post will fail with a `nil` username and password.

Reply:
my understanding of reading auth-source.el is that it'll return a plist either from one of the backends, or prompt the user if `:create` is true; The only way to quit out would be keyboard-quit which would abort the entire login flow. This is at least the case for netrc backend, and I've never seen anyone customize auth-sources in the wild or provide a custom backend